app/Libraries/OsuAuthorize.php

```diff
@@ -787,42 +792,9 @@ public function checkChatChannelJoin(?User $user, Channel $channel) : string
 
         $this->ensureCleanRecord($user, $prefix);
 
-        // FIXME: needs further check before allowing other types.
-        if (false) {
-            switch ($channel->type) {
-                case Channel::TYPES['public']:
-                    return 'ok';
-
-                case Channel::TYPES['private']:
-                    $commonGroupIds = array_intersect(
-                        $user->groupIds(),
-                        $channel->allowed_groups
-                    );
-
-                    if (count($commonGroupIds) > 0) {
-                        return 'ok';
-                    }
-                    break;
-
-                case Channel::TYPES['spectator']:
-                case Channel::TYPES['temporary']: // this and the comparisons below are needed until bancho is updated to use the new channel types
-                    if (starts_with($channel->name, '#spect_')) {
-                        return 'ok';
-                    }
-
-                    if (starts_with($channel->name, '#mp_')) {
-                        $matchId = intval(str_replace('#mp_', '', $channel->name));
-
-                        if (in_array($user->user_id, Match::findOrFail($matchId)->currentPlayers(), true)) {
-                            return 'ok';
-                        }
-                    }
-                    break;
-
-                case Channel::TYPES['multiplayer']:
-                    return 'ok';
-                break;
-            }
+        // allow joining of 'tournament' matches (for lazer/tournament client)
+        if ($channel->isBanchoMultiplayerChat() && $channel->multiplayerMatch && $channel->multiplayerMatch->isTournamentMatch()) {
```

Comment:
`optional(...->...Match)->isTournamentMatch()`? ...Do we even need the `isBanchoMultiplayerChat()` here?